### ISSUE TITLE
Documentation for almost_equals/equals is not correct

### ISSUE THREAD

**Issue by koshy1123** [geopandas/geopandas#626]
The binary predicate check `almost_equals` is actually `geom_almost_equals`. The [documentation](http://geopandas.org/reference.html#GeoSeries.almost_equals) indicates just `almost_equals`. It results in the error:
```
AttributeError: 'GeoSeries' object has no attribute 'almost_equals'
```
Some related notes:
- I'm noticing that in the [reference.rst](https://github.com/geopandas/geopandas/blob/master/doc/source/reference.rst) on GH that it's referenced as `geom_almost_equals`.
- `equals()` and `geom_equals()` behaves differently. The documentation only indicates the former and that `equals()` should be a per-object comparison returning a `bool` Series but I'm seeing just one `bool` as an output (like an aggregate) whereas `geom_equals` returns a `bool` Series. Is that expected behavior?

EDIT: Using Python 3.6 and Geopandas 0.3.0

**Comment by kuanb** [comment]
Can you print the output of dir(foo) where foo is your geoseries? Also what version of Python and Geopandas?

**Comment by koshy1123** [comment]
^updated with versions of python and geopandas
I can confirm that the dir output has `equals()`, `geom_almost_equals()`, `geom_equals()` and NO `almost_equals()`. Full list [here](https://gist.github.com/koshy1123/18138ff54bbb0271191669a0d0707604)

**Comment by jorisvandenbossche** [comment]
Yes, I think this is an error in the docs. PR to fix it very welcome!

> equals() and geom_equals() behaves differently. The documentation only indicates the former and that equals() should be a per-object comparison returning a bool Series but I'm seeing just one bool as an output (like an aggregate) whereas geom_equals returns a bool Series. Is that expected behavior?

This is because `GeoSeries.equals` is inherited from `Series.equals` (which just returns True/False depending on whether the objects are equal). And that is the reason that those predicates are called 'geom_equals' and 'geom_almost_equals' instead of 'equals' and 'almost_equals', because 'equals' was already taken.

**Comment by jorisvandenbossche** [comment]
> Yes, I think this is an error in the docs. PR to fix it very welcome!

Hmm, as you linked to the master doc (https://github.com/geopandas/geopandas/blob/master/doc/source/reference.rst), it seems this is already fixed on master. 
Can you check in the latest docs http://geopandas.readthedocs.io/en/latest/reference.html to be sure ?

**Comment by koshy1123** [comment]
Ah yes - looks like it's fixed in the latest docs - I was looking at v0.3.0 or maybe more specifically the one here: http://geopandas.org/reference.html (which I would've thought defaulted to the latest)

**Comment by jorisvandenbossche** [comment]
Yes, that page is defaulting to the latest released, the 'latest' (the link I showed) is the development version.

Closing this then
